external-crates/move/move-compiler/src/diagnostics/mod.rs

```diff
@@ -57,12 +57,16 @@ pub struct Diagnostic {
 #[derive(PartialEq, Eq, Hash, Clone, Debug, Default)]
 pub struct Diagnostics {
     diagnostics: Vec<Diagnostic>,
+    // diagnostics filtered in user code
+    filtered_user_diagnostics: Vec<Diagnostic>,
     severity_count: BTreeMap<Severity, usize>,
 }
 
 #[derive(PartialEq, Eq, Clone, Debug)]
-/// Used to filter out diagnostics, specifically used for warning suppression
-pub struct WarningFilters(BTreeMap<ExternalPrefix, UnprefixedWarningFilters>);
+/// Used to filter out diagnostics, specifically used for warning suppression. The second field
+/// determines if this filter was created to filter warnings in the user code (false) or library
+/// code (true).
+pub struct WarningFilters(BTreeMap<ExternalPrefix, UnprefixedWarningFilters>, bool);
```

Comment:
It might be nice to give these fields names now
```suggestion
pub struct WarningFilters {
  filters: BTreeMap<ExternalPrefix, UnprefixedWarningFilters>, 
  from_library: bool, // if false, it is from the user
}
```

Reply:
And what does library code here mean?

Reply:
We are filtering out all warnings that are present  in dependencies (aka library code). It would be weird to report these in the stats and in order to avoid reporting them we need to somehow know if the filtering is happening for user code or for library code. Considering how we do filtering, it seemed most sense to encode this in `WarningFilters`.

Reply:
Gotcha, I will have some new comments then :)